### ISSUE TITLE
Problem after updating to latest version with CryptoRNG

### ISSUE THREAD

**Issue by mqaa** [daegalus/dart-uuid#103]
I updated to the latest version of dart-uuid and now getting exceptions. Turns out that the genrateInternal function of the CryptoRNG class is the cause if the exception. The following line of code is the problem:

```
var k = _secureRandom.nextInt(1 << 32);
```

The problem is that 1 << 32 returns 0 and results in the error:

```
RangeError: max must be in range 0 < max ≤ 2^32, was 0
```

This can easely be checked by running the following code in DartPad:

```
void main() {
    print(1 << 32);
}
```

I had a quick look at the history of the file and this was changed by the 'Faster CryptoRNG' commit of oct 16th. The code used to be:

```
b[i] = _secureRandom.nextInt(256);
```

The following is how I use the uuid in my code for a web app:

```
import 'package:uuid/data.dart';
import 'package:uuid/rng.dart';
import 'package:uuid/uuid.dart';

class GuidGenerator {
  static const uuid = Uuid();

  String generate() {
    return uuid.v4(
      options: {
        'rng': CryptoRNG(),
      },
    );
  }
}
```

Is CrypoRNG no longer supported for web apps, am I doing something stupid or is this a bug in the uuid code?

**Comment by mqaa** [comment]
Did some more testing and this is only a problem when running the web version of the app. When running on the iOS simulator all is working fine. So it looks like it is running into some limit of javascript.

**Comment by adrianjagielak** [comment]
Temporary fix:

Create the following class:

```dart
/// See: https://github.com/daegalus/dart-uuid/issues/103
class ActuallyWorkingRNG extends RNG {
  static final _secureRandom = Random.secure();

  @override
  Uint8List generateInternal() {
    final b = Uint8List(16);

    for (var i = 0; i < 16; i++) {
      b[i] = _secureRandom.nextInt(256);
    }

    return b;
  }
}
```

Now use the `ActuallyWorkingRNG()` instead `CryptoRNG()` when initializing Uuid (e.g. `Uuid(goptions: GlobalOptions(ActuallyWorkingRNG()))`, you don't need to specify the rng each time you call v4())

**Comment by mqaa** [comment]
That is how I fixed it for now. But I think this needs to be fixed in the package because it is marked as usable for web but it is not at the moment.

**Comment by adrianjagielak** [comment]
> That is how I fixed it for now. But I think this needs to be fixed in the package because it is marked as usable for web but it is not at the moment.

Totally agree. Such basic core package should never break that catastrophically on any platform.

**Comment by daegalus** [comment]
My apologies the delay, I have been on paternity leave for my second child and it has been busy. I will try to get a fix in sometime this week

**Comment by fiurthorn** [comment]
on the web the highest Int Value is [MAX_SAFE_INTEGER](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER?retiredLocale=de).

**Comment by daegalus** [comment]
this should now be fixed in 4.2.2+
